src/cci/cas_cci.c

```diff
@@ -712,14 +712,29 @@ cci_disconnect (int mapped_conn_id, T_CCI_ERROR * err_buf)
 
       get_last_error (con_handle, err_buf);
     }
-  else if (con_handle->broker_info[BROKER_INFO_CCI_PCONNECT] && hm_put_con_to_pool (con_handle->id) >= 0)
+  else if (con_handle->broker_info[BROKER_INFO_CCI_PCONNECT] && hm_put_con_to_pool_exceed () == 0)
     {
-      cci_end_tran_internal (con_handle, CCI_TRAN_ROLLBACK);
-      API_ELOG (con_handle, 0);
+      MUTEX_LOCK (con_handle_table_mutex);
+      if (con_handle->id < 0 || hm_put_con_to_pool (con_handle->id) < 0)
+	{
+	  qe_con_close (con_handle);
 
+	  error = CCI_ER_BIND_INDEX;
```

Comment:
In the no. 739 line **'else'**, the return value of **qe_con_close()** is set to error, but in this flow inadequate index seems so definitely error(or negative value, but there seems so no negative value related error message. So, I set to error as CCI_ER_BIND_INDEX only) and I set to error CCI_ER_BIND_INDEX.

Reply:
I understand the intention. However, this error appears to be an error code related to bind parameter. If you look up the function where this error code is used, I think you can figure out the meaning of this error code.

Reply:
Did you mean that bind parameter is this one?

>"Bind parameters—also called dynamic parameters or bind variables—are an alternative way to pass data to the database. Instead of putting the values directly into the SQL statement, you just use a placeholder like ?, :name or @name and provide the actual values using a separate API call."

If then, I misunderstood that message is just out of range because in cci_get_err_msg_internal() function, there is written below;

>"Parameter index is out of range"

I will add additional error code or find something else.
if I will add one, it can be related with [APIS-947](http://jira.cubrid.org/browse/APIS-947)